schema/crdb/dbinit.sql

```diff
@@ -1480,32 +1480,36 @@ CREATE TABLE IF NOT EXISTS omicron.public.ip_pool (
     time_modified TIMESTAMPTZ NOT NULL,
     time_deleted TIMESTAMPTZ,
 
-    /* Identifies if the IP Pool is dedicated to Control Plane services */
-    internal BOOL NOT NULL,
-
     /* The collection's child-resource generation number */
     rcgen INT8 NOT NULL,
 
+    is_default BOOLEAN NOT NULL DEFAULT FALSE,
+
     /*
-     * Fields representating association with a silo or project. silo_id must be
-     * non-null if project_id is non-null. When project_id is non-null, silo_id
-     * will (naturally) be the ID of the project's silo. Both must be null if
-     * internal is true, i.e., internal IP pools must be fleet-level pools.
+     * Fields representating association with a silo or project. silo_id must
+     * be non-null if project_id is non-null. silo_id is also used to mark an IP
+     * pool as "internal" by associating it with the oxide-internal silo.
      */
     silo_id UUID,
     project_id UUID,
 
     -- if silo_id is null, then project_id must be null
```

Comment:
Does it ever make sense to have a non-NULL `project_id` and `silo_id` where the `silo_id` doesn't match the project's silo? If not, I think adjusting this so that at most one of them is non-null might be preferable. A `project_id` already uniquely identifies a project and therefore the silo it belongs to.

Reply:
True. My initial reaction was that if `silo_id` wasn't there, the query for the default pool would have to be more complicated.

```rs
dsl::ip_pool
    .filter(dsl::silo_id.eq(authz_silo_id).or(dsl::silo_id.is_null()))
    .filter(
        dsl::project_id.eq(project_id).or(dsl::project_id.is_null()),
    )
    .filter(dsl::is_default.eq(true))
    .filter(dsl::time_deleted.is_null())
    // this will sort by most specific first, i.e.,
    //
    //   (silo, project)
    //   (silo, null)
    //   (null, null)
    //
    // then by only taking the first result, we get the most specific one
    .order((
        dsl::project_id.asc().nulls_last(),
        dsl::silo_id.asc().nulls_last(),
    ))
    .select(IpPool::as_select())
    .first_async::<IpPool>(self.pool_authorized(opctx).await?)
    .await
    .map_err(|e| public_error_from_diesel_pool(e, ErrorHandler::Server))
```

But if at most one of the two was non-null, we could do something like

```sql
where default = true and (
  silo_id = authz_silo_id 
  or project_id = input_project_id
  or (silo_id is null and project_id is null)
)
```

and that should get us the same at-most-3 results. If anything the proposed modification is easier to understand rather than harder.